### PR TITLE
Add a spark-shell smoke test to premerge and nightly [databricks]

### PR DESCRIPTION
**Pull request by gerashegalov**
Contributes to #5704

This PR aims to catch issues like #9500. It modifies run_pyspark_from_build mostly to avoid recreating the logic of figuring out jar location etc.

Currently it may not catch this if do not have Spark 3.5.0 CI yet. But this is how it could reproduce the #9500 

```Bash
$ SPARK_HOME=~/dist/spark-3.1.1-bin-hadoop3.2 SPARK_SHELL_SMOKE_TEST=1 ./integration_tests/run_pyspark_from_build.sh
...
+ grep -F 'res0: Array[org.apache.spark.sql.Row] = Array([4950])'
res0: Array[org.apache.spark.sql.Row] = Array([4950])
+ echo 'SUCCESS spark-shell smoke test...'
SUCCESS spark-shell smoke test
$ echo $?
0

$ SPARK_HOME=~/dist/spark-3.5.0-bin-hadoop3 SPARK_SHELL_SMOKE_TEST=1 ./integration_tests/run_pyspark_from_build.sh
$ echo $?
1

SPARK_SHELL_SMOKE_TEST=1 \
  PYSP_TEST_spark_shuffle_manager=com.nvidia.spark.rapids.spark311.RapidsShuffleManager \
  SPARK_HOME=~/dist/spark-3.1.1-bin-hadoop3.2 \
  ./integration_tests/run_pyspark_from_build.sh
+ echo 'SUCCESS spark-shell smoke test'
SUCCESS spark-shell smoke test
$ echo $?
0

SPARK_SHELL_SMOKE_TEST=1 \
  PYSP_TEST_spark_shuffle_manager=com.nvidia.spark.rapids.spark350.RapidsShuffleManager \
  SPARK_HOME=~/dist/spark-3.5.0-bin-hadoop3 \
  ./integration_tests/run_pyspark_from_build.sh
$ echo $?
1
```

Signed-off-by: @gerashegalov

### CODE REVIEW

**Comment by gerashegalov** [comment]
build

File: integration_tests/run_pyspark_from_build.sh [hunk 1]
```diff
@@ -309,7 +309,24 @@ EOF
...
+    if [[ "$SPARK_SHELL_SMOKE_TEST" != "0" ]]; then
+        echo "Running spark-shell smoke test..."
+        # NOTE grep is used not only for checking the output but also
+        # to workaround the fact that spark-shell catches all failures.
+        # In this test it exits not because of the failure but because it encounters
+        # an EOF on stdin and injects a ":quit" command. Without a grep check
+        # the exit code would be success 0 regardless of the exceptions.
+        #
+        <<< 'spark.range(100).agg(Map("id" -> "sum")).collect()' \
+            "$SPARK_HOME"/bin/spark-shell \
```
Comment (tgravescs) [inline integration_tests/run_pyspark_from_build.sh:322]: So without the fix we put in (https://github.com/NVIDIA/spark-rapids/pull/9500), will this just hang forever?

Comment (gerashegalov) [reply integration_tests/run_pyspark_from_build.sh:322]: No I demonstrated in the description that it exits with the error exit code 1. Probably around 20sec or something until it fails

**Review by tgravescs — DISMISSED** [review]

**Comment by gerashegalov** [comment]
added shuffle manger conf to spark shell, PTAL

**Review by tgravescs — APPROVED** [review]

**Comment by gerashegalov** [comment]
build

**Comment by gerashegalov** [comment]
build

**Pull request merged**